src/primitives/line.jl

```diff
@@ -19,3 +19,14 @@ Line(a::Tuple, b::Tuple) = Line(Point(a), Point(b))
 paramdim(::Type{<:Line}) = 1
 
 (l::Line)(t) = l.a + t * (l.b - l.a)
+
+function Base.in(p::Point, l::Line)
+  w = norm(l.b-l.a)
+  d = evaluate(Euclidean(), p, l)
+  # d ≈ 0.0 will be too precise, and d < atol{T} can't scale.
+  d+w ≈ w
```

Comment:
Can you comment a bit more about the scale issue you mention? I like that you are writing very clean and concise code 💯 👍🏽 Just double checking that you are aware of `isapprox(d, zero(T), atol = atol(T))`.

Reply:
For `Sphere` example, if we change whole size, then the thresold will change together.
```julia
julia> Point(0.0,1+1e-8) in Sphere(Point(0.0,0.0),1.0)
false

julia> Point(0.0,1+1e-9) in Sphere(Point(0.0,0.0),1.0)
true

julia> Point(0.0,(1+1e-8)/100) in Sphere(Point(0.0,0.0),1/100)
false

julia> Point(0.0,(1+1e-9)/100) in Sphere(Point(0.0,0.0),1/100)
true
```

I thought this prpperty is important, because we will deal with both large and small scale.
Here's the example for `Line`:

```julia
julia> Point(0.0,1e-7) in Line(Point(0.0,0.0),Point(1.0,0.0))
false

julia> Point(0.0,1e-8) in Line(Point(0.0,0.0),Point(1.0,0.0))
true

julia> Point(0.0,1e-7/100) in Line(Point(0.0,0.0),Point(1/100,0.0))
false

julia> Point(0.0,1e-8/100) in Line(Point(0.0,0.0),Point(1/100,0.0))
true
```

If we use `isapprox(d, zero(T), atol = atol(T))`, then the thresold parameter will be fixed for specific lengh.

Reply:
The following result seems weird, but this is inevitable, I guess.
```julia
julia> Point(0.0,1e-8) in Line(Point(0.0,0.0),Point(1.0,0.0))
true

julia> Point(0.0,1e-8) in Line(Point(0.0,0.0),Point(1/100,0.0))
false
```